src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -105,14 +105,19 @@ public MessageCacheItem getMessages(Iterator<Locale> fallbackLocalesIter) {
     	MessageCacheItem cacheItem = null;
     	if (cacheService.isContainComponent()) { // Item is in cache
     		cacheItem = cacheService.getCacheOfComponent();
-    		if (cacheItem.isExpired()) { // cacheItem has expired
+    		if (cacheItem.getCachedData().isEmpty()) { // This means that the data to be used is from a fallback locale.
+				// If expired, try to first create and store cacheItem for the requested locale in a separate thread.
+				if (cacheItem.isExpired())
+					this.createCacheItemTask(fallbackLocalesIter);
```

Comment:
For creating second time, it doesn't need to do fallback. How about using 'resreshCache' without locale fallback? This will improve performance.